src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -169,59 +210,81 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
+        builder.field(fieldName, schema);
         break;
       }
 
       // FLOAT is, confusingly, double precision and effectively the same as DOUBLE. See REAL
       // for single precision
       case Types.FLOAT:
       case Types.DOUBLE: {
-        if (optional) {
-          builder.field(fieldName, Schema.OPTIONAL_FLOAT64_SCHEMA);
+        if (hasDefault) {
+          double parsedDefault = Double.parseDouble(defaultValue);
+          schema = SchemaBuilder.float64().defaultValue(parsedDefault).build();
         } else {
-          builder.field(fieldName, Schema.FLOAT64_SCHEMA);
+          schema = (optional) ? Schema.OPTIONAL_FLOAT64_SCHEMA : Schema.FLOAT64_SCHEMA;
         }
+        builder.field(fieldName, schema);
         break;
       }
 
       case Types.NUMERIC:
         if (mapNumerics) {
           int precision = metadata.getPrecision(col);
           if (metadata.getScale(col) == 0 && precision < 19) { // integer
-            Schema schema;
             if (precision > 9) {
-              schema = (optional) ? Schema.OPTIONAL_INT64_SCHEMA :
-                      Schema.INT64_SCHEMA;
+              if (hasDefault) {
+                long parsedDefault = Long.parseLong(defaultValue);
+                schema = SchemaBuilder.int64().defaultValue(parsedDefault).build();
+              } else {
+                schema = (optional) ? Schema.OPTIONAL_INT64_SCHEMA : Schema.INT64_SCHEMA;
+              }
             } else if (precision > 4) {
-              schema = (optional) ? Schema.OPTIONAL_INT32_SCHEMA :
-                      Schema.INT32_SCHEMA;
+              if (hasDefault) {
+                int parsedDefault = Integer.parseInt(defaultValue);
+                schema = SchemaBuilder.int32().defaultValue(parsedDefault).build();
+              } else {
+                schema = (optional) ? Schema.OPTIONAL_INT32_SCHEMA : Schema.INT32_SCHEMA;
+              }
             } else if (precision > 2) {
-              schema = (optional) ? Schema.OPTIONAL_INT16_SCHEMA :
-                      Schema.INT16_SCHEMA;
+              if (hasDefault) {
+                short parsedDefault = Short.parseShort(defaultValue);
+                schema = SchemaBuilder.int16().defaultValue(parsedDefault).build();
+              } else {
+                schema = (optional) ? Schema.OPTIONAL_INT16_SCHEMA : Schema.INT16_SCHEMA;
+              }
             } else {
-              schema = (optional) ? Schema.OPTIONAL_INT8_SCHEMA :
-                      Schema.INT8_SCHEMA;
+              if (hasDefault) {
+                byte parsedDefault = Byte.parseByte(defaultValue);
+                schema = SchemaBuilder.int8().defaultValue(parsedDefault).build();
+              } else {
+                schema = (optional) ? Schema.OPTIONAL_INT8_SCHEMA : Schema.INT8_SCHEMA;
+              }
             }
             builder.field(fieldName, schema);
             break;
           }
         }
 
+      //TODO: ???
       case Types.DECIMAL: {
```

Comment:
@bellemare @jfb-flipp should I handle this case?

Reply:
Isn't a decimal just a double? Wouldn't that work? Could use a float too, but if the length is unknown I would prefer a double.

Reply:
Decimals are fixed point. Looks like it's the same as a NUMERIC(M, D) to MySQL

Reply:
We might as well handle it, yeah.